### PR TITLE
docs: restore work-avoidance page

### PR DESCRIPTION
**Pull request by mikutas**
Signed-off-by: mikutas <23391543+mikutas@users.noreply.github.com>

https://argoproj.github.io/argo-workflows/work-avoidance/ (404)

Is it deleted accidentally (in #8562) ?

### CODE REVIEW

File: docs/work-avoidance.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+> v2.9 and after
+
+You can make workflows faster and more robust by employing **work avoidance**. A workflow that utilizes this is simply a workflow containing steps that do not run if the work has already been done. This simplest way to do this is to use **marker files**.
+
+Use cases:
+
+* An expensive step appears across multiple workflows - you want to avoid repeating them.
+* A workflow has unreliable tasks - you want to be able resubmit the workflow.
+
```
Comment (juliev0) [inline docs/work-avoidance.md:11]: "you want to be able to resubmit"

File: docs/work-avoidance.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+A **marker file** is a file on that indicates the work has already been done, before doing the work you check to see if the marker has already been done:
+
```
Comment (juliev0) [inline docs/work-avoidance.md:13]: nitpick: replace comma with semi-colon or period

Comment (juliev0) [reply docs/work-avoidance.md:13]: and remove "on"

**Comment by juliev0** [comment]
It appears this is where it happened: https://github.com/argoproj/argo-workflows/commit/c2f450622d93dd18ddcba8aee85543b9d020a831#diff-cb9fc269fdf9cc076fc7f14092fe63f69f0275d24e3d2e8b41c8f1ce01b3314b

There was a reorganization to the page so I think it must have accidentally been removed then. (let us know if you see anything else this might have happened to there)

By the way, your Lint error should be fixed in master now.

**Review by juliev0 — CHANGES_REQUESTED** [review]
commented on some minor typos

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**